core/src/main/scala/org/apache/spark/sql/execution/columnar/impl/JDBCSourceAsColumnarStore.scala

```diff
@@ -320,9 +327,44 @@ class JDBCSourceAsColumnarStore(override val connProperties: ConnectionPropertie
 final class ColumnarStorePartitionedRDD(
     @transient private val session: SnappySession,
     private var tableName: String,
+    @(transient @param) partitionPruner: => Int,
     @transient private val store: JDBCSourceAsColumnarStore)
     extends RDDKryo[Any](session.sparkContext, Nil) with KryoSerializable {
 
+  private[this] var allPartitions: Array[Partition] = _
+  private val evaluatePartitions: () => Array[Partition] = () => {
+    store.tryExecute(tableName, conn => {
+      val resolvedName = ExternalStoreUtils.lookupName(tableName,
+        conn.getSchema)
+      val region = Misc.getRegionForTable(resolvedName, true)
+      partitionPruner match {
+        case -1 if allPartitions != null =>
+          allPartitions
+        case -1 =>
+          allPartitions = session.sessionState.getTablePartitions(
+            region.asInstanceOf[PartitionedRegion])
+          allPartitions
+        case bucketId: Int =>
+          if (java.lang.Boolean.getBoolean("DISABLE_PARTITION_PRUNING")) {
```

Comment:
Change this system property to a proper conf variable as mentioned in a comment above.